packages/docs/src/components/DocSiteNav.tsx

```diff
@@ -1,9 +1,87 @@
 import React from 'react';
+import { VerticalNav } from '@cmsgov/design-system';
+import { VerticalNavItemProps } from '@cmsgov/design-system/src/components/VerticalNav/VerticalNavItem';
+import { useStaticQuery, graphql } from 'gatsby';
 
+interface NavItem {
+  id: string;
+  name: string;
+  relativeDirectory: string;
+  relativePath: string;
+}
+
+/**
+ * DocSiteNav
+ * queries all MDX files and generates the proper format to create a vertical nav
+ * @returns {React Element}
+ * @todo figure out which item is currently selected and mark & expand appropriately
+ * @todo Determine URL string for each item once content is integrated
+ */
 const DocSiteNav = () => {
+  const data = useStaticQuery(graphql`
+    query SiteNavQuery {
+      allFile(
+        filter: { absolutePath: {}, extension: { eq: "mdx" } }
```

Comment:
I'm not well-versed in graphql, but I'm seeing `allMdx` as an option next to `allFile`. Would the be a more efficient query to start with since you're filtering the mdx files?

Reply:
I'm now well-versed either 😅 
Looking into it, it seems like `allMdx` is used to help bring back the content of MDX files along with some other information. The one piece of info that seems to be harder to retrieve in this query is the `relativeDirectory`. I think it's possible to include, but several layers deep.
Also related to Patrick's comment, I don't see a way to group by relativeDirectory, so I'm not sure if this query is both more efficient AND gives all the required info -- looking into it!